plaidml2/op/lib/ops.cc

```diff
@@ -2034,15 +2049,41 @@ Value reshape(const Value& value) {
   return Value{edsl::reshape(I, O_dims)};
 }
 
+Value scale_gradient(const Value& value) {
+  IVLOG(1, "scale_gradient");
+  auto args = value.as_tuple();
+  if (args.size() != 2) {
+    throw std::runtime_error("scale_gradient expects 2 arguments");
+  }
+  auto I = ident(args[0].as_tensor());  // Copy for safe gradient override
+  Tensor scale;
+  if (args[1].is_float()) {
+    // Cast scale to Tensor if it's given as a float
+    scale = Tensor{args[1].as_float()};
+  } else {
+    scale = ident(args[1].as_tensor());  // Copy for safe gradient override
+  }
+  auto O = I;  // Forward pass is NoOp
+  TensorDeriv deriv = [](const Tensor& Y, const Tensor& DY, const std::vector<Tensor>& X) {
+    return std::vector<Tensor>{X[1] * DY, Tensor{0.0}};
+  };
+  auto Overridden = OverrideGrads(deriv, std::vector<Tensor>{I, scale}, O);
+  return Value{Overridden};
+}
+
 Value sigmoid(const Value& value) {
   IVLOG(1, "sigmoid");
   auto args = value.as_tuple();
   if (args.size() != 1) {
     throw std::runtime_error("sigmoid expects 1 argument");
   }
-  auto I = args[0].as_tensor();
+  auto I = ident(args[0].as_tensor());  // Copy for safe gradient override
   auto O = 1.0 / (1.0 + exp(-I));
-  return Value{O};
+  TensorDeriv deriv = [](const Tensor& Y, const Tensor& DY, const std::vector<Tensor>& X) {  //
```

Comment:
So since we implemented `sigmoid` in the op lib, we can probably remove this: https://github.com/plaidml/plaidml/blob/master/plaidml2/edsl/derivs.cc#L149